### ISSUE TITLE
Volume module is very slow on systems with PulseAudio

### ISSUE THREAD

**Issue by lahwaacz** [i3/i3status#4]
The use of asoundlib in `print_volume.h` results in very poor performance on Linux systems with PulseAudio as the default sound server. I've done a [profiling test](https://github.com/lahwaacz/i3status/commits/profiling) to support my claims, the results are [here](https://github.com/lahwaacz/i3status/blob/profiling/callgrind.out.30029) (viewable with e.g. [kcachegrind](http://kcachegrind.sourceforge.net/html/Home.html)). In summary, during 20 iterations there are 20 calls to `print_volume`, 20 calls to `pa_mainloop_run`, subsequently 3680 calls to `pa_mainloop_iterate`, 120302 calls to `pa_xmalloc` etc.

This may be either a bug in `print_volume.c`, in asoundlib itself or merely a fact showing the need for a native PulseAudio module for i3status -- I don't really know.

For the record, here is my `/etc/asound.conf` file:
```
# Use PulseAudio by default
pcm.!default {
  type pulse
  fallback "sysdefault"
  hint {
    show on
    description "Default ALSA Output (currently PulseAudio Sound Server)"
  }
}

ctl.!default {
  type pulse
  fallback "sysdefault"
}
```

**Comment by stapelberg** [comment]
You’ve stumbled upon this before. I think the library we’re using _shells out_ when you’re using PulseAudio.

If you could send a pull request to address this, that’d be great. Ideally, you’d modify the code in such a way that we don’t need to change the configuration file at all, and it’d just more or less directly talk to PulseAudio when appropriate.

**Comment by stapelberg** [comment]
This has been fixed with #7.
